src/cpp/src/cache_state_dumper.hpp

```diff
@@ -83,7 +83,7 @@ class CacheStateDumper {
      */
     void dump_cache_state(const Scheduler &schdl, const std::vector <SequenceGroup::Ptr> &sequence_groups,
                           size_t dump_count) {
-        dump_cache_state(schdl.m_block_manager, sequence_groups, dump_count);
+        dump_cache_state(*schdl.m_block_manager, sequence_groups, dump_count);
```

Comment:
maybe we can fix signature to accept shared_ptr? in this case we don't have to use `*` and it will look more natural

Reply:
Done